aws/src/main/java/org/frankframework/filesystem/AmazonS3FileSystem.java

```diff
@@ -472,53 +540,55 @@ public String getName(S3Object f) {
 
 	@Override
 	public String getParentFolder(S3Object f) {
-		int lastSlashPos = f.getKey().lastIndexOf('/');
-		return lastSlashPos > 1 ? f.getKey().substring(0, lastSlashPos) : null;
+		int lastSlashPos = f.key().lastIndexOf('/');
+		return lastSlashPos > 1 ? f.key().substring(0, lastSlashPos) : null;
 	}
 
 	@Override
 	public String getCanonicalName(S3Object f) {
-		return f.getBucketName() + BUCKET_OBJECT_SEPARATOR + f.getKey();
+		return bucketName + BUCKET_OBJECT_SEPARATOR + f.key();
 	}
 
 	@Override
 	public long getFileSize(S3Object f) {
-		updateFileAttributes(f);
-		return f.getObjectMetadata().getContentLength();
+		return getFileAttributes(f).contentLength();
```

Comment:
Gaat hij nu voor elke metadata methode (filesize, mtime etc) een losse http-call uitvoeren?

Reply:
Klopt, maar dat deed hij voorheen ook al via `updateFileAttributes`. Zie bijvoorbeeld ook `org.frankframework.filesystem.sftp.SftpFileSystem#getFileSize`, daar werkt het op een vergelijkbare manier.

Reply:
Zeker niet, `updateFileAttributes` vult 1 maal de meta-data mocht die er niet zijn. Vervolgens wordt daar informatie uit gelezen. `getFileAttributes` doet elke keer een losse call.

Reply:
Opgelost in eeb7e1d